### PR TITLE
FOUNDATIONS: Configuration - RemoveConfigurationByIdAsync

### LINKED ISSUE #151: FOUNDATIONS: Configuration - DeleteConfigurationAsync

### PR DESCRIPTION
**Pull request by pratikkaje**
Closes #151

### CODE REVIEW

File: GitFyle.Core.Api.Tests.Unit/Services/Foundations/Configurations/ConfigurationServiceTests.Exceptions.RemoveById.cs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+            var expectedConfigurationDependencyException =
+                new ConfigurationDependencyException(
+                    message: "Configuration dependency error occurred, contact support.",
+                    innerException: failedStorageConfigurationException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectConfigurationByIdAsync(someConfigurationId))
+                    .ThrowsAsync(sqlException);
+            // when
```
Comment (cjdutoit) [inline GitFyle.Core.Api.Tests.Unit/Services/Foundations/Configurations/ConfigurationServiceTests.Exceptions.RemoveById.cs:38]: Add line above

File: GitFyle.Core.Api.Tests.Unit/Services/Foundations/Sources/SourceServiceTests.Exceptions.RemoveById.cs [hunk 2]
```diff
@@ -67,7 +67,7 @@ await Assert.ThrowsAsync<SourceDependencyException>(
         }
 
         [Fact]
-        private async Task ShouldThrowDependencyValidationExceptionOnRemoveByIdIfDbConcurrencyOccursAndLogItAsync()
+        public async Task ShouldThrowDependencyValidationExceptionOnRemoveByIdIfDbConcurrencyOccursAndLogItAsync()
```
Comment (cjdutoit) [inline GitFyle.Core.Api.Tests.Unit/Services/Foundations/Sources/SourceServiceTests.Exceptions.RemoveById.cs:70]: var lockedSourceException =
                new LockedSourceException(
                    message: "Locked source record error occurred, please try again.",
                    innerException: dbUpdateConcurrencyException,
                    data: dbUpdateConcurrencyException.Data);

File: GitFyle.Core.Api.Tests.Unit/Services/Foundations/Configurations/ConfigurationServiceTests.Exceptions.RemoveById.cs (hunk 1 above)
Comment (pratikkaje) [reply GitFyle.Core.Api.Tests.Unit/Services/Foundations/Configurations/ConfigurationServiceTests.Exceptions.RemoveById.cs:38]: Added the required line break. Thanks.

File: GitFyle.Core.Api.Tests.Unit/Services/Foundations/Sources/SourceServiceTests.Exceptions.RemoveById.cs (hunk 2 above)
Comment (pratikkaje) [reply GitFyle.Core.Api.Tests.Unit/Services/Foundations/Sources/SourceServiceTests.Exceptions.RemoveById.cs:70]: Resolved. Thank you.

**Comment by pratikkaje** [comment]
> @pratikkaje update your local, and resolve open comments please.

Changes are done. Thanks.

**Review by glhays — APPROVED** [review]

**Pull request merged**